### PR TITLE
[Merged by Bors] - feat(Analysis/Analytic/IsolatedZeros): the principle of isolated zeros with `Filter.codiscreteWithin`

### PR DESCRIPTION
**Pull request by Command-Master**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Analytic/IsolatedZeros.lean [hunk 1]
```diff
@@ -235,6 +235,20 @@ theorem eqOn_zero_of_preconnected_of_frequently_eq_zero (hf : AnalyticOn 𝕜 f
...
+  simp only [ne_eq, eventually_iff, mem_codiscreteWithin, disjoint_principal_right, not_forall]
+    at nh
+  obtain ⟨x, hx, hx2⟩ := nh
+  apply eqOn_zero_of_preconnected_of_frequently_eq_zero hf hU hx
+  intro nh
+  apply hx2
+  filter_upwards [nh] with a
+  intro
+  simp_all
+
```
Comment (j-loreaux) [inline Mathlib/Analysis/Analytic/IsolatedZeros.lean:251]:
```suggestion
  simp only [or_iff_not_imp_right, ne_eq, eventually_iff, mem_codiscreteWithin, 
    disjoint_principal_right, not_forall]
  rintro ⟨x, hx, hx2⟩
  refine hf.eqOn_zero_of_preconnected_of_frequently_eq_zero hU hx fun nh ↦ hx2 ?_
  filter_upwards [nh] with a ha
  simp_all
  
```

**Review by j-loreaux — COMMENTED** [review]
Is there a higher dimensional version of this, as with `Analytic On.eqOn_of_preconnected_of_frequently_eq_zero`? I suspect not.

File: Mathlib/Analysis/Analytic/IsolatedZeros.lean [hunk 2]
```diff
@@ -235,6 +235,15 @@
...
 
+theorem eqOn_zero_or_eventually_ne_zero_of_preconnected (hf : AnalyticOn 𝕜 f U)
+    (hU : IsPreconnected U) : EqOn f 0 U ∨ ∀ᶠ x in codiscreteWithin U, f x ≠ 0 := by
+  simp only [or_iff_not_imp_right, ne_eq, eventually_iff, mem_codiscreteWithin, 
+    disjoint_principal_right, not_forall]
+  rintro ⟨x, hx, hx2⟩
+  refine hf.eqOn_zero_of_preconnected_of_frequently_eq_zero hU hx fun nh ↦ hx2 ?_
+  filter_upwards [nh] with a ha
+  simp_all
+
```
[the remark this answers is not in the document]
Comment (Command-Master) [reply Mathlib/Analysis/Analytic/IsolatedZeros.lean:246]: bot fix style

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**